bindings/wasm/bindings.d.ts

```diff
@@ -5,29 +5,140 @@ type SimplePolygon = Vec2[];
+   */
   rotate(v: Vec3): Manifold;
+
+  /**
+   * Scale this Manifold in space. This operation can be chained. Transforms are
+   * combined and applied lazily.
+   *
+   * @param v The vector to multiply every vertex by per component.
+   */
   scale(v: Vec3): Manifold;
+
+  /**
+   * Boolean union
+   */
   add(other: Manifold): Manifold;
+
+  /**
+   * Boolean difference
+   */
   subtract(other: Manifold): Manifold;
+
+  /**
+   * Boolean intersection
+   */
   intersect(other: Manifold): Manifold;
+
+  /**
+   * Increase the density of the mesh by splitting every edge into n pieces. For
+   * instance, with n = 2, each triangle will be split into 4 triangles. These
+   * will all be coplanar (and will not be immediately collapsed) unless the
+   * Mesh/Manifold has halfedgeTangents specified (e.g. from the Smooth()
+   * constructor), in which case the new vertices will be moved to the
+   * interpolated surface according to their barycentric coordinates.
+   *
+   * @param n The number of pieces to split every edge into. Must be > 1.
+   */
   refine(n: number): Manifold;
 }
 
-declare function cube(size: Vec3, center?: boolean): Manifold;
+/**
+ * Constructs a unit cube (edge lengths all one), by default in the first
+ * octant, touching the origin.
+ *
+ * @param size The X, Y, and Z dimensions of the box.
+ * @param center Set to true to shift the center to the origin.
+ */
+declare function cube(size?: Vec3, center?: boolean): Manifold;
 
+/**
+ * A convenience constructor for the common case of extruding a circle. Can also
+ * form cones if both radii are specified.
+ *
+ * @param height Z-extent
+ * @param radiusLow Radius of bottom circle. Must be positive.
+ * @param radiusHigh Radius of top circle. Can equal zero. Default is equal to
+ * radiusLow.
+ * @param circularSegments How many line segments to use around the circle.
+ * Default is calculated by the static Defaults.
+ * @param center Set to true to shift the center to the origin. Default is
+ * origin at the bottom.
+ */
 declare function cylinder(
-    height: number, radiusLow: number, radiusHigh?: number,
-    circularSegments?: number, center?: boolean): Manifold;
+  height: number, radiusLow: number, radiusHigh?: number,
+  circularSegments?: number, center?: boolean): Manifold;
 
+/**
+* Constructs a geodesic sphere of a given radius.
+*
+* @param radius Radius of the sphere. Must be positive.
+* @param circularSegments Number of segments along its
+* diameter. This number will always be rounded up to the nearest factor of
+* four, as this sphere is constructed by refining an octahedron. This means
+* there are a circle of vertices on all three of the axis planes. Default is
+* calculated by the static Defaults.
+*/
 declare function sphere(radius: number, circularSegments?: number): Manifold;
 
+/**
+ * Constructs a manifold from a set of polygons by extruding them along the
+ * Z-axis.
+ *
+ * @param crossSection A set of non-overlapping polygons to extrude.
```

Comment:
Just found out that we may have this error in both our js and c++ documentation. I think this should be polygon with holes instead of a set of non-overlapping polygons?

Reply:
I wouldn't call it an error, but there may be a clearer way to say it. I define overlapping [here](https://github.com/elalish/manifold/wiki/Manifold-Library#overlaps), so holes are not considered overlaps. I used to have polygon-with-holes, but that makes it sound like there should only be one outer contour.